src/Mailgun/Api/Pagination.php

```diff
@@ -22,12 +22,52 @@
     abstract protected function safeDeserialize(ResponseInterface $response, $className);
 
     /**
+     * @param mixed $response
+     *
+     * @return mixed|null
+     */
+    public function getPaginationNext($response)
```

Comment:
We should still try to type hint this. 
What do you think of adding an interface that cover the `get*Url()` functions from the trait?